client/src/app/shared/clock.service.ts

```diff
@@ -0,0 +1,10 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ClockService {
+  now(): Date {
```

Comment:
A simple abstraction to uncouple the dependency on the system clock with `new Date()`